code/lib/builder-vite/src/plugins/mdx-plugin.ts

```diff
@@ -46,19 +48,35 @@ export function mdxPlugin(options: Options): Plugin {
       if (!filter(id)) return undefined;
 
       const { compile } = await import('@storybook/mdx2-csf');
+      const addons = await options.presets.apply('addons', [] as StorybookConfig['addons']);
+      const docsOptions =
+        // @ts-expect-error - not sure what type to use here
+        addons.find((a) => [a, a.name].includes('@storybook/addon-docs'))?.options ?? {};
+
+      const storiesFile = isStorybookMdx(id);
+
+      const { mdxPluginOptions } = docsOptions;
 
       const mdxLoaderOptions = await options.presets.apply('mdxLoaderOptions', {
+        skipCsf: !storiesFile,
+        ...mdxPluginOptions,
         mdxCompileOptions: {
           providerImportSource: '@storybook/addon-docs/mdx-react-shim',
+          ...mdxPluginOptions.mdxCompileOptions,
+          remarkPlugins: [remarkSlug, remarkExternalLinks].concat(
```

Comment:
Should we move these plugins to mdx2-csf? 

Seems to be a couple issues asking for the removal of it https://github.com/storybookjs/storybook/issues/9329